core/scripts/lobster/extender.py

```diff
@@ -0,0 +1,173 @@
+import csv
+import random
+
+from collections import defaultdict, namedtuple
+from datetime import datetime, time, timedelta
+
+from models.message import Message, MessageType
+from lobster.parser import LobsterMessageParser
+
+
+Placement = namedtuple('Placement', ['index', 'message'])
+
+
+def _endless_copies(l):
+    i = 0
+    while True:
+        i += 1
+        for m in l:
+            yield i, m.copy()
+
+
+def _rand_range(left, right):
+    return random.randrange(left, right) if left < right else left
+
+
+def _mix_by_index(base, mix):
+    j = 0
+    l = len(mix)
+    placement = mix[j]
+    for i, b in enumerate(base):
+        while j < l and placement.index <= i:
+            yield placement.message
+            j += 1
+            placement = mix[j] if j < l else Placement(-1, -1)
+        yield b
+
+
+def weekdays(start: float, n):
+    current = datetime.fromtimestamp(start / 10 ** 9).date()
+    for _ in range(n):
+        yield datetime.combine(current, time()).timestamp() * 10 ** 9
+        offset = 3 if current.weekday() == 4 else 1
+        current = current + timedelta(days=offset)
+
+
+def _initial_qty_for_messages(messages):
+    """Calculates inital quantity for an order based on the messages that affect it.
+    If the order has a NEW_ORDER, return its quantity
+    Otherwise, sum up the executions, cancels and deletes"""
+    total = 0
+    for _, m in messages:
+        if m.message_type == MessageType.NEW_ORDER:
+            return m.share_quantity
+        elif m.message_type == MessageType.EXECUTE:
+            total += m.share_quantity
+        elif m.message_type == MessageType.DELETE:
+            return total + m.share_quantity
+        elif m.message_type == MessageType.MODIFY:
+            total += m.share_quantity
+
+    # Order has no NEW or DELETE, we will generate a delete for the same amount
+    return total * 2
```

Comment:
could just set total to proper value and break for the two above cases, so there's a single return

Reply:
Actually here trying to use a single return would require an additional boolean flag to determine if a `NEW_ORDER` or `DELETE` was seen, as line 62 is only reached if `messages` only contains executes or modifies. This case requires doubling the summed up total.
I think adding breaks here would simply make the function more confusing and make maintenance more difficult.